language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/util/rename/RenameUtil.java

```diff
@@ -128,6 +133,12 @@ private RenameUtil() {
         if (onImportDeclarationNode(context, nodeAtCursor)) {
             return Collections.emptyMap();
         }
+        
+        if (nodeAtCursor.kind() == SyntaxKind.SIMPLE_NAME_REFERENCE) {
+            if (isSelfClassSymbol(context)) {
+                return Collections.emptyMap();
+            }
+        }
 
```

Comment:
Let's merge the two conditions and also, we can merge the particular with the condition at line 133.